File: spotmybands/api/views/spotify_auth.py
import base64
import json
import requests

from django.http import HttpResponse, JsonResponse
from django.conf import settings
from django.views.generic import View


class SpotifyAuthView(View):
    def get(self, request):
        b64Key = base64.b64encode('{}:{}'.format(
            settings.SPOTIFY_CLIENT_ID, settings.SPOTIFY_CLIENT_SECRET).encode()
        )
        headers = {
            'Authorization': 'Basic {}'.format(b64Key.decode())
        }
        data = {
            'grant_type': 'client_credentials'
        }
        r = requests.post(
            'https://accounts.spotify.com/api/token', headers=headers, data=data)
        try:
            response_data = json.loads(r.text)
            access_token = response_data['access_token']
        except:
            pass
        return JsonResponse({'access_token': access_token})

File: src/server/spotifyAuth.ts
import fetch from 'node-fetch';
import { KEYS } from './utils';

export const auth = async () => {
  const authString = `${KEYS.SPOTIFY_CLIENT_ID}:${KEYS.SPOTIFY_CLIENT_SECRET}`
  const b64Key = Buffer.from(authString).toString('base64')
  console.log('base64', b64Key)
  const headers = {
    Authorization: `Basic ${b64Key}`,
    'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
  }
  const body = new URLSearchParams()
  body.append('grant_type', 'client_credentials')

  const res = await fetch(
    'https://accounts.spotify.com/api/token',
    {
      method: 'POST',
      body,
      headers,
    }
  )
  const response = await res.json()
  console.log(response)
  return response
};

export default auth;
